server/client.go

```diff
@@ -3534,6 +3534,13 @@ func (c *client) processServiceImport(si *serviceImport, acc *Account, msg []byt
 	// so we only lock once.
 	to, _ = si.acc.selectMappedSubject(to)
 
+	oreply, oacc := c.pa.reply, c.acc
+	c.pa.reply = nrr
+	if !si.isSysAcc {
+		c.mu.Lock()
+		c.acc = si.acc
```

Comment:
For some reason, I find this scary.. And if I look down the code, I see that we pass si.acc to processMsgResults/sendMsgToGateways. So is it that somewhere deep into those calls there is a reference to c.acc where we should have instead use the passed account? Would that not be safer to fix those places?

Reply:
We needed to do something like this for daisy chaining service imports together that involved the $SYS account like JS. The bug was allowing JS to work.

I will keep working on this with followup PRs as well since complete behavior is not correct just yet for full import of JS.